### ISSUE TITLE
Week 1: Toolkit

### ISSUE THREAD

**Issue by bradreeder** [foundersandcoders/master-reference#24]
The proposal from FAC9 is that week 1 should be organised around providing the student a toolkit they'll need for the rest of the curriculum. Themes include:

* Pair programming
* Git workflow
* User stories
* Code reviewing

Related #18 (high priority issue)

**Comment by bradreeder** [comment]
Other notes from FAC9's feedback for week 1 include:

* Rename project to Portfolio, not blog (they can't do CMS at this point.)
* Include talk or workshop on accessibility.
* More context around why we need mobile-first design.
* Replace flexbox morning challenge with something grid-system related.
* Have another talk from FAC alumni on FAC values & community ethos (Mark's talk was well received.)

FAC9's S-G-C notes for this week soon to follow.

**Comment by Cleop** [comment]
This would also include setup and clear structuring of:
- coding for everyone/women and bin rotas
- allocate business dev teams
- Using the space ie. giving out keys, buying supplies, out of hours use, watering plants.
- Regular FAC events ie. monthly business meeting, Wednesdays at Makers Meetup etc.
-The weekly routine
-Punctuality / absence

**Comment by njsfield** [comment]
Perhaps as @jsms90 suggested in #25, and @harrygfox's point about pre-requisites involving understanding of HTML & CSS before the course, semantic HTML could be left to pre-requisites and workshops could be given on accessibility, Mobile First design and perhaps a grid system workshop too. This would cover stop point [5](https://foundersandcoders.gitbooks.io/fac9/content/week1/stop_go_continue.html)

I think the project should simply be a profile page for a small team, perhaps just one page, but with focus on Mobile First, use of a grid system, semantic HTML and well-organised CSS (trickier topics like Flexbox and BEM could be introduced given in the following weeks). This would cover go point [5](Create key skills objective or each project so we know our aim is to make a sophisticated functioning blog). Groups will still be practicing git flow (and problem solving, e.g. dealing with merge conflicts in CSS files), and allocating tasks to build grid/media queries/html etc, in go point [8](better allocation to tasks)

**Comment by jsms90** [comment]
@njsfield I agree with everything in your first paragraph wholeheartedly! But I'm not sure that limiting them to one page is necessary, and I actually think that BEM really should be left in week 1.

When it comes to..a style guide (if that's the right terminology?), I think it's just about picking one and sticking to it. I'm not sure what benefit would come with introducing structure to their CSS later on. But I can see lots of benefit from having it in the course as early as possible - week 1 is the only project where they focus this heavily on HTML & CSS and will get them used to it so that they are more likely to consider this and be consistent later on.

As for flexbox, if we're going to go with the fact that they should learn to write CSS without flexbox because of the lack of support on IE (and Safari?) then you're probably right about getting them to do week 1 without flexbox! We would probably need to bring this up directly, as I know that many people at the meetups use it. But I see no problem with laying out this philosophy of "vanilla" everything :)

**Comment by bradreeder** [comment]
Week 1 S-G-C notes (avoided repeats):

##### Stop

* Ensure it's emphasised that the projects are team-focused - no solo programming in week 1. Better allocation of tasks.
* More on git branching and gitflow before start of projects. Maybe add git udacity to precourse.
* Never make more than one template (?)

##### Go 

* Workshops on website architecture (grids, columns etc) and mobile-first.
* Create key skills objective for each project.
* Create repo to share notes/ resources. Keep good notes. Flashcards channel.
* Ensure expectations are set early re: washing up after yourself.
* Distinguish between QA and code review.

##### Continue

* Pair programming & swapping pairs
* Encourage showcasing new tools/features/code
* Learning by doing (projects)
* Forcing people to use git
* Leaving at 6

**Comment by des-des** [comment]
Hi all I have added @Jbarget and @nogainbar as they are teaching this week in Nazareth

**Comment by njsfield** [comment]
# Core Learning Outcomes

- FAC values
- Responsibilities & house rules
- CSS - BEM, Mobile-First, Responsive Web Design
- Semantic HTML
- Gitflow and use of github
- Accessibility 
- Pair programming
- Group working

## Monday
- 10.00 - 11.00 -- Introduction & Name game
- 11.00 - 11.30 -- Talk from Ines
- 11.30 - 13.00 -- Course overview, Gitbook overview, FAC repositories for project work.
- 14.00 - 14.30 -- Pair Programming demonstration from alumni with known kata, including 10 minutes Q & A.
- 14.30 - 15:00 -- Refactoring Exercise 1. Refactor HTML to semantic HTML in pairs.
- 15:00 - 15:15 — Toilet/Coffee Break
- 15:15 - 15:45 — Talk on Accessibility
-  15:45 - 16:30 — Refactoring Exercise 2. Refactor HTML with accessibility standards
- 17.00 - 18.00 -- Talk (Mark).

##  Tuesday
- 10:00 - 12:00 -- DBS Check Induction
- 12:00 - 13:00 -- Git flow walkthrough (with Heather and Gabriel)
- 14.00 - 15:30 -- Research Projects
- 15:30 - 17:00 -- Review and Presentations
- 17:00 - 18:00 -- Project introduction

##  Wednesday
- 10 - 11 — Morning Challenge (Build Grid System with BEM)
- 11 - Projects

## Thursday
- Projects…

## Friday
- 10-11 -- Code review
- 11-13 -- Respond to issues
- 13-14 -- Stop/go/continue
- 14-5 -- Presentations
- 5-6 -- upstairs projects



## Research Project Topic Ideas

### Command Line
- Customise command line (oh my zsh etc)
- Install a program from source

### GitHub  (project: refactor yesterdays project)
- Document git project & best practices
- Labels/Milestones

### Accessibility … ?
- … ?

### CSS (project: refactor yesterdays project)
- Responsive v mobile First Design
- BEM

**Comment by bradreeder** [comment]
Looking noice!

* Timings for Friday seem off, as SGC is covering the time we normally have lunch? Plus remember team retrospectives!
* Don't think project introduction needs an hour on Tuesday, may as well extend research a bit longer?
* Are we proposing having business development on Tuesday morning re: DBS induction? Fine by me, might just be worth inducting them to the idea of business dev at the same time so they know what's upcoming.

**Comment by jsms90** [comment]
@njsfield @Cleop I just got a reply from Stemnet, who are happy to come in on Monday 4-6pm for the induction. I think @bradreeder is right, and that we are better off introducing them to a coherent week structure right form the start

We probably can't have such a long coffee/toilet break:
> 15:00 - 15:15 — Toilet/Coffee Break

but let's still factor one in.

I'm happy to try and squeeze the accessibility workshop into 15:00-16:00 (no separate talk, just a short intro to the topic and then straight into the workshop itself)?


Then we could change to Tuesday to:
- 10:00 - 11:00 -- Morning Challenge (Build Grid System with BEM)
- 11:00 - 12:45 -- Git flow workshop
- 12:45 - 13:00 -- Introduce research topics & whiteboard jigsaw teams
LUNCH
- 14.00 - 15:30 -- Research Projects
- 15:30 - 17:00 -- Review and Presentations
- 17:00 - 18:00 -- Project introduction

Thoughts?

**Comment by bradreeder** [comment]
Related to this 

> Personal note: one thing I'd like is if you could encourage students to understand during induction the open-source & communal nature of our resources. The resources belong to the community, they're as much theirs to improve. I.e. if they run into problems with workshops they should feel welcome & encouraged to raise issues so we know how to improve them.

I wouldn't mind being involved in the 11:30-13:00 course overview talk. I'm happy for it to be a short slot -- I think it would just be good that I get to introduce myself, and that they understand how the reforms we're implementing relates to them, such as the above.

**Comment by jsms90** [comment]
@bradreeder Absolutely!

@iteles Would you be able to point me to those accessibility resources we were talking about please?

**Comment by Cleop** [comment]
# Core Learning Outcomes

- FAC values
- Responsibilities & house rules
- CSS - BEM, Mobile-First, Responsive Web Design
- Semantic HTML
- Gitflow and use of github
- Accessibility 
- Pair programming
- Group working

## Monday
- 10.00 - 11.00 — Introduction & Name game
- 11.00 - 11.30 — Welcome talk from Ines
- 11.30 - 13.00 — Course overview, Gitbook overview, FAC repositories for project work.
- 14.00 - 14.30 — Pair Programming demonstration from alumni with known kata, including 10 minutes Q & A.
- 14.30 - 15:00 — Refactoring Exercise 1. Refactor HTML to semantic HTML in pairs.
- 15:00 - 15:05 — Toilet/Coffee Break
- 15:05 - 16:00 — Accessibility Workshop 2. Refactor HTML with accessibility standards
- 16.00 - 18.00 — DBS Check Induction

##  Tuesday
- 10:00 - 11:00 — Morning Challenge
- 11:00 - 13:00 — Git flow walkthrough
- 14.00 - 15:30 — Research Projects
- 15:30 - 17:00 — Review and Presentations
- 17:00 - 18:00 — Project introduction

##  Wednesday
- 10 - 11 — Morning Challenge (Build Grid System with BEM)
- 11 - Projects

## Thursday
- Projects…
- 17.00 - 18.00 — Talk (Mark).

## Friday
- 10-11 — Code review
- 11-13 — Respond to issues
- 13-14 — Stop/go/continue
- 14-5 — Presentations
- 5-6 — upstairs projects



## Research Project Topic Ideas

### Command Line
- Customise command line (oh my zsh etc)
- Install a program from source

### GitHub  (project: refactor yesterdays project)
- Document git project & best practices
- Labels/Milestones

### Accessibility … ?
- … ?

### CSS (project: refactor yesterdays project)
- Responsive v mobile First Design
- BEM


**When to introduce Grid system and how long does it need?

**Comment by Cleop** [comment]
Task Strategy going forward:

- ReadMe topics (All)

- Git flow walkthrough (Jen)
- Accessibility workshop (Jen)

- Define pair programming (Cleo)
- Course overview, Gitbook overview, FAC repositories for project work. talk (Cleo)
- Contact Mark to organise FAC values talk (Cleo)
- Contact Ines for Welcome talk (Cleo)

- Morning challenge CSS (Nick)
- Group project outline (Nick)
- Morning challenge Refactor semantic/access (Nick)

**Comment by njsfield** [comment]
[Pair Programming Resource](https://github.com/foundersandcoders/old-playbook/blob/master/pair-programming.md)

**Comment by iteles** [comment]
@nelsonic also wrote [this on code reviews](https://github.com/foundersandcoders/old-playbook/blob/master/code-reviews.md) which will be pretty useful for the first week's code reviews too! It seems to have got lost in the merge (#68 ) but it should be brought in again.

**Comment by njsfield** [comment]
Branch created for contribution-
**week-1-toolkit**

**Comment by njsfield** [comment]
[Wednesday Morning Challenge Draft](https://github.com/njsfield/css-gallery-challenge)

**Comment by jsms90** [comment]
@iteles It's [here](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/general/pair-programming.md) in the 'general' folder. Even though we make use of it in week 1:
> 14.00 - 14.50 — Introduction to Pair Programming (1 mentor)

I think it's better to keep it in the 'general' folder, because it's such an important resource to come back to during every subsequent week :+1:

**Comment by des-des** [comment]
@iteles https://github.com/foundersandcoders/master-reference/blob/master/coursebook/general/code-reviews.md.

I do not think anything was lost in the merge!

**Comment by bradreeder** [comment]
My general feeling I'm getting from feedback on the Anna Freud design sprint and the experience of running sprint planning today, is that students should be introduced to user-journeys and user-stories as a part of project-planning before starting on client work.

If you all disagree they should be introduced in week 1 as a part of the workflow you take students through I can create a separate issue for it @jsms90 @Cleop @njsfield ?

**Comment by jsms90** [comment]
@bradreeder I definitely agree. It feels strange that we haven't done our projects like this all along, and are introduced to a different workflow now that it's real. So the earlier we introduce it to FAC10, the better imo 

@njsfield @Cleop Do you think this is something that we could do just after the morning challenge on wednesday, as part of introducing them to the projects?

I think it makes sense to talk about it after they know what the project is going to be.

**Comment by jsms90** [comment]
Just to clarify, @Cleop 's talk:
> Course overview, Gitbook overview, FAC repositories for project work. talk (Cleo)

refers to:

1. The [code of conduct](https://github.com/foundersandcoders/master-reference/blob/master/code_of_conduct.md)

2. The following files in the new [london-curriculum repo](https://github.com/foundersandcoders/london-curriculum) (as per the passing of [this curriculum committee proposal](https://github.com/foundersandcoders/london-curriculum/issues/6#issue-207825340) ):
+ Calendar
+ Facilities
+ List of people active in space
+ List of responsibilities for current mentors (@bradreeder would the contents of this include?)
+ The meetup rotas
+ Who's on each student project team

and a blank markdown file, ready to be filled with:
+ Stop-go-continue notes

3. The structure of the FAC10 github organisation

I suggest making a new organisation called `FAC10`, like the one FAC9 has: https://github.com/FAC9

Make a repo inside this organisation called `research` (not READMES), with sub folders called
`week1`
`week2`
etc.

Tell FAC10 to make each of their project repos inside the `FAC10` organisation, so they can be found easily - as per @Jbarget 's [comment](https://github.com/foundersandcoders/master-reference/issues/12#issue-194549368). Tell them to use the following naming convention so that searching is easy:
`week1-<name of team1>`
`week1-<name of team2>`
`week1-<name of team3>`
`week1-<name of team4>`
`week2-<name of team1>`
etc.

Make a `morning-challenge-solutions` repo - or should these be put into master-reference?

**Comment by jsms90** [comment]
On day 1, maybe just before they arrive, they should all be added to the `foundersandcoders` github organisation, the `FAC10` github organisation, and their new FAC10 gitter channel, inside foundersandcoders

**Comment by bradreeder** [comment]
@jsms90 @Cleop to clarify of that section how much time do I have to speak to FAC10 and are there any topics you want me to cover? 

Otherwise, I'd intend to:

* Give them background around course facilitation role
* The vision of the curriculum going forwards
* Explain a few key terms (like 'democratic education', 'learning communities') in relation to the educational model & organisation's vision and mission statements.
* Talk a little about our governing structure (if no one else is), inc. how to become a member.

I think solutions to morning challenges should go in the FAC10 organisation for now.

**Comment by jsms90** [comment]
@bradreeder I think you, @Cleop @iteles (and @markwilliamfirth ?) need to discuss who is talking about what and how much time is allotted for each thing

**Comment by jsms90** [comment]
@Jbarget @nogainbar
I guess you'll have seen [week 1's schedule](https://github.com/foundersandcoders/master-reference/blob/cdf88f24ad8a42f99fe1f2b7165cd281565edb11/coursebook/week-1/README.md) by now :)  I just wanted to update you on the workshops I've made.

### [Accessibility workshop](https://github.com/jsms90/web-accessibility)
It's nearly done!

You guys should definitely download these chrome extensions yourselves, in preparation:
+ [HTML 5 outliner](https://chrome.google.com/webstore/detail/html5-outliner/afoibpobokebhgfnknfndkgemglggomo?hl=en) - chrome extension  
+ [ChromeVox](https://chrome.google.com/webstore/detail/chromevox/kgejglhpjiefppelpmljglcjbhoiplfn?hl=en) - chrome extension  
+ [Accessibility Developer Tools](https://chrome.google.com/webstore/detail/accessibility-developer-t/fpkknkljclfencbdbgkenhalefipecmb?hl=en) - chrome extension

And remember that we are now using [various accessibility tools]() on their weekly projects [during code review](https://github.com/foundersandcoders/master-reference/issues/81). This is highlighted very clearly in the [user stories](https://github.com/foundersandcoders/master-reference/blob/c4a21927baa14575dfbf6fda1f53cb7b540f0f33/coursebook/week-1/project.md) for week 1's project :+1: 

I have to apologise for the fact that students are just going to be faced with a wall of text to read through. But I think the intro is important - most websites are made without proper consideration for accessibility concerns.

An accessibility workshop hasn't been done before (or not that I'm aware of), so I've been kind of learning this stuff as I was making it. Getting some text written up while I was learning felt like the most time-efficient way for me to get something together.

Some notes for a future iteration: 
+ I thought about giving a **brief talk** instead, and turning the content of the README into presentation slides. But day 1 is already heavy with us standing at the front and talking at the students. I want to break that up a bit - but reading a wall of text isn't exactly collaborative learning
+ It might be better to **chromecast some really short youtube clips** that express each of the relevant points. I'm trying to find something punchy in length, that we still have the required impact to help people understand how inexcusable it is do write inaccessible websites/apps

You guys (and @des-des @shiryz @esraajb): **Please let me know how it goes, running this workshop in Nazareth!** I'd be really interested to hear how students find it & how you choose to run it :)

### Git - [Natalia's "learn git basics"](https://github.com/NataliaLKB/learn-git-basics)
I had intended to make something that reflects our increasingly advanced [prerequisites](http://www.foundersandcoders.com/apply) e.g. making their own github page, and [precourse](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/precourse/resources), as well as making the learning more collaborative. But the accessibility workshop has taken much longer than I'd anticipated, and I thought it was more important to focus on the one I had to make from scratch. Plus, Natalia's workshop is still awesome!! :grin: 

Note: based on my cohort's s-g-c "Go" point:
> More on git branching and gitflow before start of projects. Maybe add git udacity to precourse

we updated the [git & github learning outcomes](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/precourse/learning-outcomes.md#git-and-github) on the precourse. And we added [the first part](https://github.com/foundersandcoders/master-reference/blob/master/coursebook/precourse/resources.md#git--github) of Udacity to the precourse.

FAC10 have definitely had enough practice with the most basic git commands. I think it might be worth jumping straight into the [git flow](https://github.com/NataliaLKB/learn-git-basics#github-flow) section, then going back to [the basic tutorial](https://github.com/NataliaLKB/learn-git-basics#tutorial) to cement that, and then taking them through more advanced commands ([not just `git add`, `git commit`, `git push`, etc](https://github.com/NataliaLKB/learn-git-basics/blob/master/cheatsheet.md))

I'm pretty familiar with FAC10's level BUT I have no idea about Naz-FAC1! So you may want to start from the beginning, rather than skipping the intro? **Again, please let me know how this goes on your end! It will be interesting to see whether this 2 hours needs to be done quite differently, based on the different prereqs/precourse stage at each location.**

**Comment by jsms90** [comment]
@njsfield @Cleop Could you add the relevant links to our README.md when you get the chance:
https://github.com/foundersandcoders/master-reference/pull/117#issue-208645061

**Comment by Jbarget** [comment]
@jsms90 thanks for all of that! @nogainbar and i are going to be going through it all tonight

**Comment by des-des** [comment]
@jsms90 we decided to go for https://github.com/FACN1. I can do a just go and do a full replace

**Comment by jsms90** [comment]
### Suggested format of README.md file inside `research day` repo:

#### Week 1 (A bit unusual because of [week 1's research topics])
+ [Command Line](nicelinkhere.com)
+ Git - [Website](nicelinkhere.com) - [Before](nicelinkhere.com) | [After](nicelinkhere.com)
+ Accessibility - [Website](nicelinkhere.com) - [Before](nicelinkhere.com) | [After](nicelinkhere.com)
+ CSS - [Website](nicelinkhere.com) - [Before](nicelinkhere.com) | [After](nicelinkhere.com)

#### Week 2 (based on [this](https://github.com/foundersandcoders/master-reference/blob/c3f5e4c3f63f88fa2895a0029a015f118041ed28/coursebook/week-2/research-afternoon.md))
+ [Unit vs Integration testing](nicelinkhere.com)
+ [Manual testing vs. Automation testing](nicelinkhere.com)
+ [Technical Spikes](nicelinkhere.com)
+ [QUnit Assertions](nicelinkhere.com)
+ [BlanketJS](nicelinkhere.com)

#### Week 3

#### Week 4

#### Week 5
etc.

Students would need to keep this readme updated as part of research day, by adding the relevant links to the markdown files they will be creating inside the `week-1`, `week-2`, etc folders (see my [folder structure suggestion](https://github.com/foundersandcoders/master-reference/issues/24#issuecomment-280699826))

For week 1, the idea behind the "before" & "after" links is for them to look back on the progress that they made when they're planning week 1 for the next cohort :crossed_fingers: 
+ [Website](nicelinkhere.com) - would link directly to the _hosted_ version of the website they've chosen

+ [Before](nicelinkhere.com) - should take you to the _github repo_ of the website **as it was before changes were made i.e. the right commit in the commit history of the repo**.

+ [After](nicelinkhere.com) - would go to the github repo as it was after changes were made i.e. the right commit in the commit history of the repo **because they might continue working on it in future, so just linking to the github repo itself won't be very helplful**.

**Comment by bradreeder** [comment]
Week delivered. Thanks everyone!
